src/main/java/org/springframework/data/redis/aot/RedisRuntimeHints.java

```diff
@@ -112,24 +141,31 @@ public void registerHints(RuntimeHints hints, @Nullable ClassLoader classLoader)
 				TypeReference.of(DecoratedRedisConnection.class));
 
 		// keys are bound by a proxy
-		boundOperationsProxy(BoundGeoOperations.class, hints);
-		boundOperationsProxy(BoundHashOperations.class, hints);
-		boundOperationsProxy(BoundKeyOperations.class, hints);
-		boundOperationsProxy(BoundListOperations.class, hints);
-		boundOperationsProxy(BoundSetOperations.class, hints);
-		boundOperationsProxy(BoundStreamOperations.class, hints);
-		boundOperationsProxy(BoundValueOperations.class, hints);
-		boundOperationsProxy(BoundZSetOperations.class, hints);
+		boundOperationsProxy(BoundGeoOperations.class, classLoader, hints);
+		boundOperationsProxy(BoundHashOperations.class, classLoader, hints);
+		boundOperationsProxy(BoundKeyOperations.class, classLoader, hints);
+		boundOperationsProxy(BoundListOperations.class, classLoader, hints);
+		boundOperationsProxy(BoundSetOperations.class, classLoader, hints);
+		boundOperationsProxy(BoundStreamOperations.class, classLoader, hints);
+		boundOperationsProxy(BoundValueOperations.class, classLoader, hints);
+		boundOperationsProxy(BoundZSetOperations.class, classLoader, hints);
 		boundOperationsProxy(
-				TypeReference.of("org.springframework.data.redis.core.BoundOperationsProxyFactory$DefaultBoundKeyOperations"),
-				hints);
+				TypeReference.of("org.springframework.data.redis.core.BoundOperationsProxyFactory$BoundKeyOperations"),
```

Comment:
Isn't this a typo?
Doesn't look that there is an inner class like this, but that `DefaultBoundKeyOperations` is really there.
I believe @wilkinsona 's point (as well as mine) to not have this proxy hint at all. The one for `BoundKeyOperations` should be enough. No? 
Maybe reflection for those methods in the  `BoundKeyOperations` has to be exposed?
Might not be related, but I used `@Reflective` on my interfaces' methods to give them access from SpEL...

Reply:
thanks @artembilan - refined hints.